### PR TITLE
add code of register

### PR DESCRIPTION
**Pull request by RufiMohammed**

### CODE REVIEW

**Comment by RufiMohammed** [comment]
register

File: app/Http/Controllers/Api/Auth/AuthController.php [hunk 1]
```diff
@@ -42,24 +45,131 @@ public function login(Request $request)
...
+            $result = Sign_up::selectTeam($ambassador['leader_gender'],$ambassador['gender']);
+            if ($result->count() == 0){
+                //Check Teams With Less Than 12 Members
+                $result=Sign_up::selectTeam_between($ambassador['leader_gender'],$ambassador['gender'],"1","12");
+                if ($result->count() == 0){
+                     //Check Teams With Less More 12 Members
+                     $result=Sign_up::selectTeam($ambassador['leader_gender'],$ambassador['gender'],">","12");
+                     if ($result->count() == 0){
+                       // print_r($ambassador);
+                        $ambassadorWithoutLeader = User::create($ambassador);
```
Comment (saraIsmailSE) [inline app/Http/Controllers/Api/Auth/AuthController.php:88]: Request id should be null

File: app/Models/Sign_up.php [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+
+class Sign_up extends Model
+{
+    use HasFactory;
+    public function selectHighPriority($leader_gender,$ambassador_gender){
+      if($leader_gender == 'any'){
+        $users = DB::table('leader_requests')
+        ->join('users', 'leader_requests.leader_id', '=', 'users.id')
+        ->join('high_priority_requests', 'leader_requests.id', '=', 'high_priority_requests.request_id')
```
Comment (saraIsmailSE) [inline app/Models/Sign_up.php:17]: Gender should be specified

File: app/Models/Sign_up.php [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        ->where('leader_requests.is_done', '=', 0 )
+        ->where('leader_requests.gender', '=', $ambassador_gender)
+        ->where(function($query) use ($leader_gender) {
```
Comment (saraIsmailSE) [inline app/Models/Sign_up.php:20]: Need explanation

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 4]
```diff
@@ -17,12 +17,14 @@ public function up()
...
             $table->string('name');
             $table->string('email')->unique();
+            $table->string('gender');
             $table->timestamp('email_verified_at')->nullable();
             $table->string('password');
             $table->rememberToken();
             $table->boolean('is_blocked')->default(0);
             $table->boolean('is_hold')->default(0);
             $table->boolean('is_excluded')->default(0);
+            $table->integer('request_id')->default(0);
```
Comment (saraIsmailSE) [inline database/migrations/2014_10_12_000000_create_users_table.php:27]: Legit be nullable

**Pull request merged**